src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/services_almost_circular_public.php

```diff
@@ -93,6 +95,26 @@ protected function getBar3Service()
         return $instance;
     }
 
+    /**
+     * Gets the public 'bar5' shared service.
+     *
+     * @return \stdClass
+     */
+    protected function getBar5Service()
+    {
+        $a = ${($_ = isset($this->services['foo5']) ? $this->services['foo5'] : $this->getFoo5Service()) && false ?: '_'};
+
+        if (isset($this->services['bar5'])) {
```

Comment:
shouldn't this come first, before defining the `$a` variable ?

Reply:
ah no, understand. This is the call to ` $this->getFoo5Service()` which can register it in `$this->services`